homeassistant/components/aurora_abb_powerone/sensor.py

```diff
@@ -0,0 +1,98 @@
+"""Support for Aurora ABB PowerOne Solar Photvoltaic (PV) inverter."""
+
+import logging
+import voluptuous as vol
+
+from aurorapy.client import AuroraSerialClient, AuroraError
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_ADDRESS, CONF_NAME, DEVICE_CLASS_POWER, POWER_WATT)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_COMPORT = 'rs485'
+DEFAULT_ADDRESS = 2
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_COMPORT): cv.string,
+    vol.Optional(CONF_ADDRESS, default=DEFAULT_ADDRESS): cv.positive_int,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Aurora ABB PowerOne device."""
+    devices = []
+    _LOGGER.debug("Setting up AuroraABBPowerone")
+    comport = config.get(CONF_COMPORT)
+    name = config.get(CONF_NAME, "Solar PV")
+    address = config.get(CONF_ADDRESS, 1)
+
+    _LOGGER.debug("Intitialising com port=%s address=%s", comport, address)
+    client = AuroraSerialClient(address, comport, parity='N', timeout=1)
+
+    devices.append(AuroraABBSolarPVMonitorSensor(client, name, 'Power'))
+    add_devices(devices, True)
+
+
+class AuroraABBSolarPVMonitorSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, client, name, typename):
+        """Initialize the sensor."""
+        self._name = name + " " + typename
```

Comment:
Please use new style string formatting instead of string concatenation.

Reply:
Updated in 26e669d8e